src/main/java/com/arpnetworking/tsdcore/model/CompositeUnit.java

```diff
@@ -0,0 +1,338 @@
+/**
+ * Copyright 2016 Groupon.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.tsdcore.model;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableMap;
+import net.sf.oval.constraint.NotNull;
+
+
+/**
+ * Composite Unit used in 2g file format.
```

Comment:
No. You can NOT bleed the file model into the internal model. The entire point of the design is to keep them separate so changes do NOT ripple across the codebase.